README
Kakapo

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the README
file from a module distribution so that people browsing the archive
can use it to get an idea of the module's uses. It is usually a good idea
to provide version information here so that people can decide whether
fixes for the module are worth downloading.


INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Kakapo

You can also look for information at:

    RT, CPAN's request tracker (report bugs here)
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Kakapo

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Kakapo

    CPAN Ratings
        http://cpanratings.perl.org/d/Kakapo

    Search CPAN
        http://search.cpan.org/dist/Kakapo/


LICENSE AND COPYRIGHT

Copyright (C) 2012 Joel Gomez

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; version 2 dated June, 1991 or at your option
any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

A copy of the GNU General Public License is available in the source tree;
if not, write to the Free Software Foundation, Inc.,
59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.



Changes
Revision history for Kakapo

0.01    Date/time
        First version, released on an unsuspecting world.



test.pl
#!/bin/perl -w

use Speech::Synthesis;

my @voices = Speech::Synthesis->InstalledVoices(engine => 'Festival');
foreach (@voices){
   print "\n".$_->{id};
}


lib/TTS.pm
package TTS;

use strict;
use warnings;
use Speech::Synthesis;
use File::Slurp;
use Encode;
use File::Temp qw/ tempfile/;
use Exporter;
use Switch;
use Data::Dumper;
use File qw( load_file );


our @ISA = qw(Exporter);
our @EXPORT = qw( convert );

=head1 NAME

Kakapo - The great new Kakapo!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Quick summary of what the module does.

Perhaps a little code snippet.

    use Kakapo;

    my $foo = Kakapo->new();
    ...

=head1 EXPORT

A list of functions that can be exported.  You can delete this section
if you don't export anything, such as for a purely object-oriented module.

=head1 SUBROUTINES/METHODS

=head2 function1

=cut
sub convert {
	my ( $self, $file ) = @_;
	
	load_file( $self, $file ) unless ( -e "/tmp/kakapo.tmp" );
	my $voice = "voice_".$self->{voices}->get_active_text;
	my $cmd = "text2wave /tmp/kakapo.tmp -o /home/jgomez/Escritorio/prop.wav -eval '($voice)'";
	system($cmd);
	
	unlink("/tmp/kakapo.tmp");
	$self->{vta_converted}->show;
}


=head1 AUTHOR

Joel Gomez, C<< <joelgomezb at gmail.com> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-kakapo at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Kakapo>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.




=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Kakapo


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Kakapo>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Kakapo>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Kakapo>

=item * Search CPAN

L<http://search.cpan.org/dist/Kakapo/>

=back


=head1 ACKNOWLEDGEMENTS


=head1 LICENSE AND COPYRIGHT

Copyright 2012 Joel Gomez.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; version 2 dated June, 1991 or at your option
any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

A copy of the GNU General Public License is available in the source tree;
if not, write to the Free Software Foundation, Inc.,
59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.


=cut

1; # End of Kakapo


library.pm
#!/usr/bin/perl 
use Speech::Synthesis;

package rutinas;

sub adios {
    	my $engine = 'Festival';
	my @voices = Speech::Synthesis->InstalledVoices( engine => $engine );
	my @avatars = Speech::Synthesis->InstalledAvatars( engine => $engine );
	
        my $ss = Speech::Synthesis->new(
					engine => 'Festival', 
					avatar => undef, 
					language => 'es', 
					voice => 'JuntaDeAndalucia_es_pa_diphone', 
					async => 0, 
				       );
	    $ss->speak( $_[0] || "test" );
}

sub tts {
	my $engine = 'Festival';
	my @voices = Speech::Synthesis->InstalledVoices( engine => $engine );
	my @avatars = Speech::Synthesis->InstalledAvatars( engine => $engine );
	
        my $ss = Speech::Synthesis->new(
					engine => 'Festival', 
					avatar => undef, language => 'es', 
					voice => 'JuntaDeAndalucia_es_pa_diphone', 
					async => 0, 
				       );
	    $ss->speak( $_[0] || "test" );
}

1;


Makefile.PL
use 5.006;
use strict;
use warnings;
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME                => 'Kakapo',
    AUTHOR              => q{Joel Gomez <joelgomezb@gmail.com>},
    VERSION_FROM        => 'lib/Kakapo.pm',
    ABSTRACT_FROM       => 'lib/Kakapo.pm',
    ($ExtUtils::MakeMaker::VERSION >= 6.3002
      ? ('LICENSE'=> 'gpl')
      : ()),
    PL_FILES            => {},
    PREREQ_PM => {
        'Test::More' => 0,
    },
    dist                => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean               => { FILES => 'Kakapo-*' },
);

requires 'Speech::Synthesis';
requires 'Gtk2::Ex::Simple::List';
requires 'Gtk2::GladeXML';
requires 'Speech::Synthesis';
requires 'Config::General';
requires 'File::Slurp';
requires 'Encode';
requires 'File::Slurp';
requires 'Exporter';
requires 'File::Temp';
requires 'Switch';
requires 'IO::File';


lib/File.pm
package File;

use strict;
use warnings;
use Speech::Synthesis;
use File::Slurp;
use Encode;
use IO::File;
use Exporter;
use Switch;
use Data::Dumper;


our @ISA = qw(Exporter);
our @EXPORT = qw(load_file);

=head1 NAME

Kakapo - The great new Kakapo!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Quick summary of what the module does.

Perhaps a little code snippet.

    use Kakapo;

    my $foo = Kakapo->new();
    ...

=head1 EXPORT

A list of functions that can be exported.  You can delete this section
if you don't export anything, such as for a purely object-oriented module.

=head1 SUBROUTINES/METHODS

=head2 function1

=cut
sub load_file {
	my ( $self, $file ) = @_;

	my $fh = IO::File->new("/tmp/kakapo.tmp", "w");

	switch ( $file ) {
		case  /.txt/ { txt ( $self, $file ); }
		else {	print Dumper("Formato aun no soportado!!! :("); }
	}
}

sub txt {
	my ( $self, $file ) = @_;

	my $buffer_file = Gtk2::TextBuffer->new;
			open(LISTA, $file) || die(rutinas::tts("No se pudo abrir el archivo"));
			while(<LISTA>){
				$buffer_file->insert_at_cursor( decode("utf8", $_) );
				write_file( "/tmp/kakapo.tmp", {binmode => ':utf8' }, decode ("utf8", $_ ) );
			}
			close(LISTA);	
	$self->{text}->set_buffer($buffer_file);
	$self->{ejecutar}->set_sensitive(1);
	$self->{convertir}->set_sensitive(1);
}

=head1 AUTHOR

Joel Gomez, C<< <joelgomezb at gmail.com> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-kakapo at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Kakapo>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.




=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Kakapo


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Kakapo>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Kakapo>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Kakapo>

=item * Search CPAN

L<http://search.cpan.org/dist/Kakapo/>

=back


=head1 ACKNOWLEDGEMENTS


=head1 LICENSE AND COPYRIGHT

Copyright 2012 Joel Gomez.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; version 2 dated June, 1991 or at your option
any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

A copy of the GNU General Public License is available in the source tree;
if not, write to the Free Software Foundation, Inc.,
59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.


=cut

1; # End of Kakapo


lib/Kakapo.pm
package Kakapo;

use strict;
use warnings;
use Gtk2 '-init';
use Gtk2::Ex::Simple::List;
use Gtk2::GladeXML;
use Speech::Synthesis;
use Config::General;
use File::Slurp;
use base qw( Gtk2::GladeXML::Simple );
use Encode;
use TTS;
use File qw(load_file);

=head1 NAME

Kakapo - The great new Kakapo!

=head1 VERSION

Version 0.01

=cut

our $VERSION = '0.01';


=head1 SYNOPSIS

Quick summary of what the module does.

Perhaps a little code snippet.

    use Kakapo;

    my $foo = Kakapo->new();
    ...

=head1 EXPORT

A list of functions that can be exported.  You can delete this section
if you don't export anything, such as for a purely object-oriented module.

=head1 SUBROUTINES/METHODS

=head2 function1

=cut

sub new {
  my $class = shift;
  my $self = $class->SUPER::new( File::Spec->rel2abs(File::Spec->curdir()) . '/resources/main.glade' );
  return $self;
}

sub run {
  my ( $self ) = @_;
#  $self->{vta_login}->show;
  $self->begin;
  $self->{ventana_principal}->show;
  $self->{ventana_principal}->signal_connect('delete_event' => sub { Gtk2->main_quit; });
  Gtk2->main;
}

sub begin {
	my ( $self ) = @_;

	$self->{ejecutar}->set_sensitive(0);
	$self->{convertir}->set_sensitive(0);
# tengo que verificar que el proceso de festival  esté corriendo
	my $engine = 'Festival';
	my @voices = Speech::Synthesis->InstalledVoices(engine => $engine);

# tengo que verificar la lista de voces de festival instaladas

	$self->{voices}->new_text();
	foreach (@voices) {
		$self->{voices}->append_text($_->{name});
	}

	$self->{voices}->set_active(0);


}

sub on_hablar_clicked {
	my ( $self ) = @_;
	
		my $buffer_file = Gtk2::TextBuffer->new;
		my $file = $self->{filechooserbutton1}->get_filename;
		convert( $self, $file );
}

use Data::Dumper;
sub on_filechooserbutton1_file_set {
	my $self = shift;

	my $buffer_file = Gtk2::TextBuffer->new;
	my $file = $self->{filechooserbutton1}->get_filename;
	load_file( $self, $file );
}

sub on_imagemenuitem2_activate {
	my $self = shift;
	$self->{filechooserdialog1}->show();
	print Dumper($self->{filechooserbutton1}->get_filename);
#	$self->on_filechooserbutton1_file_set();
}

sub on_bt_aceptar_clicked {
	my ( $self ) = @_;

	$self->{vta_converted}->hide;
}

sub gtk_main_quit {
  Gtk2->main_quit;
}

sub on_bcerrar_clicked {
  my $self = shift;

  $self->gtk_main_quit;
}

sub function1 {
}

=head2 function2

=cut

sub function2 {
}

=head1 AUTHOR

Joel Gomez, C<< <joelgomezb at gmail.com> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-kakapo at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Kakapo>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.




=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Kakapo


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Kakapo>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Kakapo>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Kakapo>

=item * Search CPAN

L<http://search.cpan.org/dist/Kakapo/>

=back


=head1 ACKNOWLEDGEMENTS


=head1 LICENSE AND COPYRIGHT

Copyright 2012 Joel Gomez.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; version 2 dated June, 1991 or at your option
any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

A copy of the GNU General Public License is available in the source tree;
if not, write to the Free Software Foundation, Inc.,
59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.


=cut

1; # End of Kakapo
